### ISSUE TITLE
Remove Microsoft.VisualStudio.LanguageServices layer and OOP from Codespace client

### ISSUE THREAD

**Issue by tmat** [dotnet/roslyn#49913]
Currently the VS layer binaries are loaded on the client. The VS workspace is created and it launches ServiceHub which proffers our ISB services. These are useless on the client. Not only that these services conflict with ISB instances running on the server in its ServiceHub process. 

We should
1) Create a new project (`Microsoft.CodeAnalysis.Codespaces.Client`) that builds a VSIX and contains all code that's specific to Codespaces client (such as client workspace). 
2) Update setup authoring install Microsoft.CodeAnalysis.Codespaces.Client.vsix on the client and remove other Roslyn VSIXes.

**Comment by dibarbet** [comment]
This could be a bigger work item than we initially anticipated

Razor uses the VS workspace on the client to access and create services
https://github.com/dotnet/aspnetcore-tooling/blob/f433d3d6b7cd4111160b9be0440c8c6881052803/src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/DefaultVisualStudioWorkspaceAccessor.cs#L27

And there are many other usages of it here http://index/?query=Visualstudioworkspace&rightProject=Microsoft.VisualStudio.Editor.Razor&file=VisualStudioWorkspaceAccessor.cs&rightSymbol=izuqcxhi4drq which we'll need to categorize to determine if they need it running on the client.

**Comment by dibarbet** [comment]
@tmat as a short term solution to unblock Allison while I work with other teams (and razor) to remove dependencies on the VS workspace on the client, I wonder if it would be reasonable to just disable the OOP process from starting if in a codespace client context (e.g. here http://sourceroslyn.io/#Microsoft.VisualStudio.LanguageServices/Implementation/Remote/VisualStudioWorkspaceServiceHubConnector.cs,23 or http://sourceroslyn.io/#Microsoft.CodeAnalysis.Workspaces.Test.Utilities/Remote/InProcRemoteHostClientProvider.cs,85).

**Comment by tmat** [comment]
@dibarbet Does Razor care about the VS workspace though? If it just needs services those should be available from the CloudEnvironmentClientWorkspace as well. The External Accessor can be updated to import that workspace instead, no?

**Comment by tmat** [comment]
Also, we should update the External Access APIs to not expose the workspace itself, but to only provide services (e.g. `GetService<T>()` method).

**Comment by dibarbet** [comment]
@tmat yeah we can work around it - but unfortunately the VisualStudioWorkspace type is public and is directly mef imported, it doesn't go through external access.  And a fair number of people import it - http://index/?query=Visualstudioworkspace&rightProject=Microsoft.VisualStudio.Editor.Razor&file=AuthConnectedServiceInstanceAspNetCoreHandler.cs&line=49

I'll work through these to ensure that either they aren't running on the client or we remove the import, but it just may take a bit longer.

**Comment by tmat** [comment]
I see. Forgot this is a public type :(. We'll need to work through all usages that are relevant for Nexus. Might just need to remove it in a PR and see what breaks.
For code that we already know of we should start replacing it. But you are right, this is gonna take longer.

Meanwhile let's disable remote services when running on the Codespaces client for now.
